docs/source/release.rst

```diff
@@ -167,25 +80,27 @@ Release the new version on Github
     You need to have write access to the Mitiq Github repository to make
     a new release.
 
-Make a new release on Github
+There should be a new draft release created by the tag you made in the previous step
 `here <https://github.com/unitaryfund/mitiq/releases>`__.
 
-    - Choose the tag you recently created, and add information on the release
-      by pulling from ``CHANGELOG.md`` as in previous releases.
-    - Github will create compressed files with the repository. Upload the
-      ``mitiq.pdf`` file and add the locally generated distribution tarball and
-      wheel.
+    - Github will create compressed files with the repository. 
```

Comment:
Does one need to publish the draft release?

Reply:
Yup, added a note.